progs/jubjub.py

```diff
@@ -219,3 +220,31 @@ def __str__(self):
 
     async def double(self):
         return self
+
+
+async def share_mul(context: Mpc, p: Point, x: list) -> SharedPoint:
+    """
+    The multiplication of the share of a field element and a point
+    e.g. [x]G -> [X], where G is a point on the given elliptic curve
+    x is the bitwise shared value,
+    starting from the most significant bit.
+
+    Source: Section `Scalar multiplication` of
+    https://andrea.corbellini.name/2015/05/17/elliptic-curve-cryptography-a-gentle-introduction/ # noqa: E501
+    """
+    if isinstance(p, Ideal):
+        return SharedIdeal(p.curve)
+
+    product = Ideal(p.curve)
+    addend = p
+    x.reverse()
+    one_share = context.Share(1)
+
+    equalities = await asyncio.gather(*[equality(context, i_, one_share) for i_ in x])
+    opened_equalities = await asyncio.gather(*[eq.open() for eq in equalities])
```

Comment:
this is not correct, it is revealing each bit of the secret x, by comparing each bit of x to 1 and then opening the equality share. Think of another way to achieve this without using an `if` statement!